src/libos/src/fs/inode_file.rs

```diff
@@ -233,6 +233,14 @@ impl INodeFile {
         if access_mode.writable() && inode.metadata()?.type_ == FileType::Dir {
             return_errno!(EISDIR, "Directory cannot be open to write");
         }
```

Comment:
The check is at line 239

Reply:
from https://man7.org/linux/man-pages/man2/open.2.html

**If the file already exists and is a regular file and the
              access mode allows writing (i.e., is O_RDWR or O_WRONLY)
              it will be truncated to length 0.  If the file is a FIFO
              or terminal device file, the O_TRUNC flag is ignored.
              Otherwise, the effect of O_TRUNC is unspecified.**